stubs/docker/docker/api/image.pyi

```diff
@@ -39,9 +40,9 @@ class ImageApiMixin:
         repository: str,
         tag: str | None = None,
         stream: bool = False,
-        auth_config: Incomplete | None = None,
+        auth_config: dict[str, Any] | None = None,
```

Comment:
This matches the type for `auth_config` we use in `get_registry_data`. In future, I think we can make this a `TypedDict`.